src/main/java/wromaciej/hvac_sim/thermo/Ids.java

```diff
@@ -0,0 +1,28 @@
+package wromaciej.hvac_sim.thermo;
+
+import wromaciej.hvac_sim.display.Item;
+
+import java.util.HashMap;
+import java.util.Set;
+
+public class Ids {
+
+    private static HashMap<Integer, Item> items;
+
+    private static int maximalId;
```

Comment:
it would be tough lecture, but read about multithreading and synchronised blocks, volatile fields.

you should definitely make maximalId volatile, and getUniqueId synchronised to defend against concurrency. YOu could even make maximalId AtomicInteger: http://tutorials.jenkov.com/java-util-concurrent/atomicinteger.html